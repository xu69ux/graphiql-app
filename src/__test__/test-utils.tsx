import { render, RenderOptions } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { ReactElement } from 'react';
import { LanguageContext } from '../contexts/LanguageContext';

export const setLanguage = jest.fn();

export const MockLanguageProvider = ({
  children,
}: {
  children: ReactElement;
}) => {
  const mockContextValue = {
    language: 'eng',
    setLanguage,
  };

  return (
    <LanguageContext.Provider value={mockContextValue}>
      <Router>{children}</Router>
    </LanguageContext.Provider>
  );
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: MockLanguageProvider, ...options });

export * from '@testing-library/react';
export { customRender as render };
