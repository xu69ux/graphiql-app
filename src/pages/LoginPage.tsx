import { useEffect } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { FormLogIn } from '../components';
import { translations } from '../contexts/translations';
import useLanguage from '../hooks/useLanguage';

import '@styles/Auth.css';

export const LoginPage = () => {
  const navigate = useNavigate();
  const userIs = sessionStorage.getItem('authInfo');
  const { language } = useLanguage();

  useEffect(() => {
    if (userIs) navigate('/graphiql');
  }, [userIs, navigate]);

  return (
    !userIs && (
      <div className='auth-container' data-testid='login-page'>
        <h1 className='auth-title' data-testid='login-title'>
          {translations?.[language]?.loginTitle}
        </h1>
        <FormLogIn />
        <p className='no-account'>
          {translations?.[language]?.noAccount}
          <Link to='/signup' className='signup-link' data-testid='signup-link'>
            {translations?.[language]?.signup}
          </Link>
          !
        </p>
      </div>
    )
  );
};
