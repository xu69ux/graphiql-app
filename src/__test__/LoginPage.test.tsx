import { act, render, screen } from './test-utils';
import LoginPage from '../pages/LoginPage';

describe('LoginPage component', () => {
  test('renders correctly', async () => {
    await act(async () => {
      const { container } = render(<LoginPage />);
      expect(container).toMatchSnapshot();
    });
  });

  test('renders log in title', async () => {
    await act(async () => {
      render(<LoginPage />);
    });
    expect(screen.getByTestId('login-title')).toBeInTheDocument();
  });

  test('renders log in form', async () => {
    await act(async () => {
      render(<LoginPage />);
    });

    expect(screen.getByPlaceholderText(/Email/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/Password/i)).toBeInTheDocument();
  });

  test('renders sign up link', async () => {
    await act(async () => {
      render(<LoginPage />);
    });
    const linkElement = screen.getByTestId('signup-link');
    expect(linkElement).toBeInTheDocument();
  });
});
