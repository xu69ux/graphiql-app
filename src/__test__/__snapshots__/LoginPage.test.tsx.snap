// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component renders correctly 1`] = `
<div>
  <div
    class="auth-container"
    data-testid="login-page"
  >
    <h1
      class="auth-title"
      data-testid="login-title"
    >
      Log in
    </h1>
    <form>
      <div
        class="login"
      >
        <div
          class="input-wrapper"
        >
          <input
            name="email"
            placeholder="email"
            type="text"
          />
        </div>
        <div
          class="error"
        />
        <div
          class="input-wrapper"
        >
          <input
            name="password"
            placeholder="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="password-indicator"
        >
          <div
            style="width: 100%; height: 100%; opacity: 0;"
          >
            <div
              style="width: 0%; height: 100%; opacity: 0;"
            />
          </div>
        </div>
        <div
          class="error"
        />
        <button
          class="btn btn-log"
          data-testid="login-button"
          type="submit"
        >
          Log in
        </button>
      </div>
    </form>
    <p
      class="no-account"
    >
      No account? Then, 
      <a
        class="signup-link"
        data-testid="signup-link"
        href="/signup"
      >
        sign up
      </a>
      !
    </p>
  </div>
</div>
`;
